src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,79 +1,241 @@
+      val initScriptStorageObject = populateBucket(googleProject, bucketName, dataprocConfig.initActionsScriptName, content)
+
+      // Create an array of all the certs, cluster docker compose file, and the site.conf for the apache proxy
+      val certs = Array(dataprocConfig.jupyterServerCrtName, dataprocConfig.jupyterServerKeyName, dataprocConfig.jupyterRootCaPemName,
+        dataprocConfig.clusterDockerComposeName, dataprocConfig.jupyterProxySiteConfName)
+      // Put the rest of the initialization files in the init bucket
+      val storageObjects = certs.map { certName => populateBucket(googleProject, bucketName, certName, new FileContent(null, new File(dataprocConfig.configFolderPath, certName))) }
+      // Return an array of all the Storage Objects
+      storageObjects :+ initScriptStorageObject
+    }
+  }
+
+  /* Upload the given file into the given bucket */
+  private def populateBucket(googleProject: GoogleProject, bucketName: String, fileName: String, content: AbstractInputStreamContent): StorageObject = {
+    // Create a storage object
+    val storageObject = new StorageObject().setName(fileName)
+    // Create a storage object insertion request
+    val fileInserter = storage.objects().insert(bucketName, storageObject, content)
+    // Enable direct media upload
+    fileInserter.getMediaHttpUploader().setDirectUploadEnabled(true)
+    try {
+      executeGoogleRequest(fileInserter) //returns a StorageObject
+    } catch {
+      case e: GoogleJsonResponseException => throw new BucketInsertionException(googleProject, bucketName)
+    }
+  }
+
+  /* Kicks off building the cluster. This will return before the cluster finishes creating. */
+  private def build(googleProject: GoogleProject, clusterName: String, clusterRequest: ClusterRequest, bucketName: String)(implicit executionContext: ExecutionContext): Future[DataprocOperation] = {
+    Future {
+      // Create a GceClusterConfig, which has the common config settings for resources of Google Compute Engine cluster instances,
+      //   applicable to all instances in the cluster. Give it the user's service account.
+      //   Set the network tag, which is needed by the firewall rule that allows leo to talk to the cluster
+      val gce = new GceClusterConfig()
+        .setServiceAccount(clusterRequest.serviceAccount)
+        .setTags(List(dataprocConfig.clusterNetworkTag).asJava)
+
+      // Create a NodeInitializationAction, which specifies the executable to run on a node.
+      //    This executable is our init-actions.sh, which will stand up our jupyter server and proxy.
+      val initActions = Seq(new NodeInitializationAction().setExecutableFile(GoogleBucketUri(bucketName, dataprocConfig.initActionsScriptName)))
+
+      // Create a SoftwareConfig and set a property that makes the cluster have only one node
+      val softwareConfig = new SoftwareConfig().setProperties(Map("dataproc:dataproc.allow.zero.workers" -> "true").asJava)
```

Comment:
Going forward, creating a cluster will create only 1 master node, instead of 2 workers